raiden_contracts/contracts/MonitoringService.sol

```diff
@@ -90,18 +90,27 @@ contract MonitoringService is Utils {
 
     /// @notice Deposit tokens used to reward MSs
     /// Can be called by anyone several times and on behalf of other accounts
-    /// @param beneficiary The account benefitting from the deposit
-    /// @param amount The amount of tokens to be depositted
-    function deposit(address beneficiary, uint amount) public {
-        require(amount > 0);
+    /// @param beneficiary The account benefiting from the deposit
+    /// @param total_deposit Idempotent function that sets the total_deposit of
+    /// tokens of the beneficiary
+    function deposit(address beneficiary, uint256 total_deposit) public
+    {
+        require(total_deposit != 0);
+
+        uint256 added_deposit;
+
+        // Calculate the actual amount of tokens that will be transferred
+        added_deposit = total_deposit - balances[beneficiary];
 
         // This also allows for MSs to deposit and use other MSs
-        balances[beneficiary] += amount;
+        balances[beneficiary] += added_deposit;
+
+        emit NewDeposit(beneficiary, added_deposit);
 
         // Transfer the deposit to the smart contract
-        require(token.transferFrom(msg.sender, address(this), amount));
+        require(token.transferFrom(msg.sender, address(this), added_deposit));
 
-        emit NewDeposit(beneficiary, amount);
+        require(balances[beneficiary] >= added_deposit);
```

Comment:
Couldn't we cover both `require(total_deposit != 0);` and `require(balances[beneficiary] >= added_deposit);` by doing `total_deposit > balances[beneficary]` instead?

Reply:
I used the pattern that was also used here in TokenNetwork.setDeposit because I was also trying to keep the checks grouped, as agreed. In this case, `require(total_deposit > balances[beneficary])` can work.

Reply:
I think `require(total_deposit > balances[beneficiary])` solves under/overflow in a short way,  not sure though which conventions Loredana refers too.

Reply:
@err508 https://github.com/raiden-network/raiden-contracts/issues/47. In `TokenNetwork.setDeposit` it wasn't so straight forward to get the participant deposit and do the check at the beginning of the function, so I opted for a check at the end. As said, for this case here, `require(total_deposit > balances[beneficiary])` does the job.

Reply:
As Loredana says, `require(total_deposit > balances[beneficiary])` does the trick and we get the same checks guarantees from one line rather than two. So please use that instead.